docs/integrations/consul/k8s.md

```diff
@@ -0,0 +1,313 @@
+            name: frontend
+            port:
+              number: 3000
+      - path: /api
+        pathType: Prefix
+        backend:
+          service:
+            name: public-api
+            port:
+              number: 8080
+```
+
+:::tip This ingress object:
+- Uses the `ngrok` ingress class
+- The host is the ngrok domain name you selected that is static
+- There is a route for `/` that routes to the frontend service on port 3000
+- There is a route for `/api` that routes to the public-api service on port 8080
+:::
+
+  Open your `$NGROK_DOMAIN_NAME` domain in your browser and see the Hashicups application!
+
+  ![Hashicups App](./img/hashicups.png)
+
+
+## **Step 5**: Add OAuth Protection to the App {#add-oauth-protection-to-the-app}
+
+Now that we have the Hashicups application running, we can add OAuth protection to it. We'll use the OAuth module of the ngrok Ingress Controller to add OAuth protection to the app. This will allow us to use Google OAuth to protect the app.
+
+1. Create an [NgrokModuleSet](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/user-guide/route-modules.md)
+
+  The NgrokModuleSet is a custom resource that allows you to configure the modules of the ngrok Ingress Controller. We'll create one to configure the OAuth module. Create the following `NgrokModuleSet`
+
+  ```yaml
+  kind: NgrokModuleSet
+  apiVersion: ingress.k8s.ngrok.com/v1alpha1
+  metadata:
+    name: oauth-module
+    namespace: consul
+  modules:
+    oauth:
+      google:
+        emailAddresses:
+        # Replace with your email address
+        - my-email@my-domain.com
+        # Or specify a list of domains instead of individual email addresses
+        # emailDomains:
+        # - my-domain.com
+  ```
+  :::tip Ngrok vs User Managed OAuth Application
+
+  This example uses the ngrok Managed OAuth Application for simplicity. To use your own Managed OAuth Application, you can [Creating a custom Google OAuth application](https://ngrok.com/docs/cloud-edge/modules/oauth/google/) or use an existing one.
+  You can then create a Kubernetes Secret with the `clientId` and `clientSecret` and reference the secret in your module set like this [example](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/user-guide/route-modules.md#user-managed-oauth-application).
+  :::
+
+1. Add the module to the ingress
+
+  Now that we have the module set created, we can add it to the ingress. We'll add an annotation to the ingress to add the module set to it.
+
+  ```bash
+  kubectl annotate ingress ingress-consul k8s.ngrok.com/modules=oauth-module -n consul
+  ```
+
+  This applies the OAuth module to each route on our ingress object. Navigate to your Hashicups app at `$NGROK_DOMAIN_NAME` and you'll see the Google OAuth screen.
+
+  ![Google OAuth Screen](./img/google-oauth.png)
+
+  Sign into your account and you'll be redirected to the Hashicups app! Now only you can order from Hashicups from anywhere!
```

Comment:
> you'll be redirected to the Hashicups app! Now only you can order from Hashicups from anywhere!

you'll be redirected to the _HashiCups_ app! Now only you can order from _HashiCups_ from anywhere!